Fixed broken execute_process() call under GNUstep shell on Windows

ElMostafaIdrassi opened the pull request:
For gnustep-config bash script to be invoked properly by `execute_process()` under GNUstep shell on Windows, the `${GNUstepConfig}` call must preceded by `sh`. Otherwise, `execute_process()` fails, leading to `objc_def_flags` and `objc_def_link_flags` being empty, with `RESULT_VARIABLE` containing : `%1 is not a valid win32 application`.

ElMostafaIdrassi committed:
Fixed broken execute_process() call under GNUstep shell on Windows

For gnustep-config bash script to be invoked properly by execute_process() under GNUstep shell on Windows, the ${GNUstepConfig} call must preceded by "sh". Otherwise, execute_process() fails, leading to "objc_def_flags" and "objc_def_link_flags" being empty, with RESULT_VARIABLE containing : "%1 is not a valid win32 application".

ElMostafaIdrassi committed:
Updated README.md with instructions on how to build using GNUstep under Windows

Building using GNUstep under Windows is broken, due to the fact that gcc, by default, makes linked libraries appear before the source, leading to the libraries not being linked as the symbols in them are not yet encountered.

mulle-nat commented:
The existing GNUstep code was written by someone else, so I can only guess if yours is correct or not (same for the old GNUstep code though). I don't want all that text in the `README.md` though. I'd put that into a separate `BUILD-GNUSTEP-WINDOWS.md` if that's OK with you,

ElMostafaIdrassi commented:
Well you can give it a try if you have some time, without adding `sh` in the `execute_process()`, the compiler and linker flags are empty on Windows, you can print them using `message( STATUS...)`, and without the manual trick, the build fails. 

**Maybe you should add a note that this was the behaviour using GNUstep shell from MSYS version 0.30 downloaded from [here](http://www.gnustep.org/windows/installer.html) and it comes with GCC 4.6.1**. 

For the Windows build steps, feel free to manage that as you wish ;)

Cheers

ElMostafaIdrassi committed:
Create BUILD-GNUSTEP-WINDOWS.md

Added build steps under Windows using CMake and GNUstep shell including a manual trick to make it work

ElMostafaIdrassi committed:
Update README.md

mulle-nat commented:
I will probably put up a new version of this and some other changes this week.
